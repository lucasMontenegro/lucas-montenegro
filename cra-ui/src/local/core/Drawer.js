import React, { Fragment } from "react"
import { makeStyles } from "@material-ui/styles"
import MuiDrawer from "@material-ui/core/Drawer"
import List from "@material-ui/core/List"
import IconButton from "@material-ui/core/IconButton"
import CloseIcon from "@material-ui/icons/Close"
import makeTranslations from "local/makeTranslations"
const navLabels = makeTranslations({
  en: `Navigation buttons`,
  es: `Botones de navegaci√≥n`,
})
const closeLabels = makeTranslations({
  en: `Close left drawer`,
  es: `Cerrar panel lateral`,
})
const useStyles = makeStyles({
  close: {
    textAlign: `right`,
  },
})
export default function Drawer ({ languageCode, width, viewState, navLinks, children }) {
  const classes = useStyles()
  const content = (
    <Fragment>
      {viewState.isMobile && (
        <div className={classes.close}>
          <IconButton
            id="close-temporary-drawer"
            aria-label={closeLabels[languageCode]}
            aria-controls="temporary-drawer"
            onClick={viewState.drawer.close}
          >
            <CloseIcon />
          </IconButton>
        </div>
      )}
      {children}
      <nav id="drawer-nav" aria-label={navLabels[languageCode]}><List>{navLinks}</List></nav>
    </Fragment>
  )
  return (
    viewState.isMobile ? (
      <MuiDrawer
        id="temporary-drawer"
        ModalProps={{ keepMounted: true }} // Better open performance on mobile.
        variant="temporary"
        open={viewState.drawer.isOpen}
        onClose={viewState.drawer.close}
        PaperProps={{ style: { width } }}
      >
        {content}
      </MuiDrawer>
    ) :
    (
      <MuiDrawer id="permanent-drawer" variant="permanent" PaperProps={{ style: { width } }}>
        {content}
      </MuiDrawer>
    )
  )
}