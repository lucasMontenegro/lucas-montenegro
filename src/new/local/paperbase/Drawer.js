import React, { Fragment } from "react"
import { makeStyles } from "@material-ui/styles"
import MuiDrawer from "@material-ui/core/Drawer"
import List from "@material-ui/core/List"
import IconButton from "@material-ui/core/IconButton"
import CloseIcon from "@material-ui/icons/Close"
import makeTranslations from "new/local/utils/makeTranslations"
import PropTypes from "prop-types"
import { languageCodePropType } from "new/local/supportedLanguages"
import { viewStatePropType } from "new/local/paperbase/useViewState"
const navLabels = makeTranslations({
  en: `Navigation buttons`,
  es: `Botones de navegaci√≥n`,
})
const closeLabels = makeTranslations({
  en: `Close left drawer`,
  es: `Cerrar panel lateral`,
})
const useStyles = makeStyles({
  close: {
    textAlign: `right`,
  },
  paper: {
    paddingLeft: 0,
    paddingRight: 0,
  },
}, { name: `Drawer` })
export default function Drawer ({ languageCode, width, viewState, navLinks, children }) {
  const classes = useStyles()
  const content = (
    <Fragment>
      {viewState.isMobile && (
        <div className={classes.close}>
          <IconButton
            id="close-temporary-drawer"
            aria-label={closeLabels[languageCode]}
            aria-controls="temporary-drawer"
            onClick={viewState.drawer.close}
          >
            <CloseIcon />
          </IconButton>
        </div>
      )}
      {children}
      <nav id="drawer-nav" aria-label={navLabels[languageCode]}><List>{navLinks}</List></nav>
    </Fragment>
  )
  const PaperProps = { style: { width } }
  const drawerClasses = { paper: classes.paper }
  return (
    viewState.isMobile ? (
      <MuiDrawer
        id="temporary-drawer"

        // ModalProps={{ keepMounted: true }}
        // Better open performance on mobile.
        // Doesn't work with portals

        classes={drawerClasses}
        variant="temporary"
        open={viewState.drawer.isOpen}
        onClose={viewState.drawer.close}
        PaperProps={PaperProps}
      >
        {content}
      </MuiDrawer>
    ) :
    (
      <MuiDrawer id="permanent-drawer" classes={drawerClasses} variant="permanent" PaperProps={PaperProps}>
        {content}
      </MuiDrawer>
    )
  )
}
Drawer.propTypes = {
  languageCode: languageCodePropType.isRequired,
  width: PropTypes.number.isRequired,
  viewState: viewStatePropType.isRequired,
  navLinks: PropTypes.node,
  children: PropTypes.node,
}