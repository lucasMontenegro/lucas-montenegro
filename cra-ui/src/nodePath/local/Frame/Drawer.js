import React from "react"
import theme from "local/theme"
import { makeStyles } from "@material-ui/core/styles"
import useMediaQuery from "@material-ui/core/useMediaQuery"
import MuiDrawer from "@material-ui/core/Drawer"
import MenuSlider from "./MenuSlider"
const leftWidth = `256px`
const drawerColor =`rgba(255, 255, 255, 0.7)`
const drawerBgColor = `#18202c`
const breakpointUp = theme.breakpoints.up(800)
const useStyles = makeStyles({
  paper: {
    outline: `none`,
    display: `flex`,
    backgroundColor: drawerBgColor,
  },
  slider: {
    height: `100vh`,
    flex: `0 0 4ch`,
  },
  children: {
    color: drawerColor,
    backgroundColor: drawerBgColor,
    height: `100vh`,
    flex: `0 1 100%`,
    overflowY: `auto`,
    WebkitOverflowScrolling: `touch`, // Add iOS momentum scrolling.
  },
})
function Drawer (props) {
  const classes = useStyles(props)
  const hidden = useMediaQuery(breakpointUp)
  if (hidden) {
    return null
  }
  return (
    <MuiDrawer
      PaperProps={{
        className: classes.paper,
        style: { width: leftWidth },
      }}
      variant="temporary"
      open={props.open}
      onClose={props.onClose}
    >
      <MenuSlider
        {...props.sliderProps}
        id="top-slider"
        className={classes.slider}
      />
      <div className={classes.children}>{props.children}</div>
    </MuiDrawer>
  )
}
export default Drawer