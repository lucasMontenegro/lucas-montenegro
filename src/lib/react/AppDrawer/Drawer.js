import { makeStyles } from "@material-ui/core/styles"
import useTranslation from "lib/react/useTranslation"
import useMediaQuery from "@material-ui/core/useMediaQuery"
import React, { Fragment } from "react"
import Toolbar from "@material-ui/core/Toolbar"
import CloseButton from "./CloseButton"
import Nav from "./Nav"
import MuiDrawer from "@material-ui/core/Drawer"
import PropTypes from "prop-types"
const desktopBreakpoint = theme => theme.breakpoints.up(`md`)
const useStyles = makeStyles(theme => ({
  flexibleSpace: {
    flex: `0 1 100%`,
  },
  paper: {
    width: theme.spacing(32),
    paddingLeft: 0,
    paddingRight: 0,
  },
  children: {
    [desktopBreakpoint(theme)]: {
      marginLeft: theme.spacing(32),
    },
  },
}), { name: `lib-react-app_drawer` })
export default function Drawer (props) {
  const { onClose } = props
  const classes = useStyles()
  const t = useTranslation()
  const isDesktop = useMediaQuery(desktopBreakpoint)
  const content = (
    <Fragment>
      <Toolbar>
        <div className={classes.flexibleSpace} />
        <CloseButton isDesktop={isDesktop} t={t} onClick={onClose} />
      </Toolbar>
      <Nav t={t} onClick={onClose} windowTitle={props.windowTitle} />
    </Fragment>
  )
  const drawerClasses = { paper: classes.paper }
  const drawer = (
    isDesktop ? <MuiDrawer classes={drawerClasses} variant="permanent">{content}</MuiDrawer> :
    (
      <MuiDrawer
        id="lib-react-app_drawer"
        ModalProps={{ keepMounted: true }}
        classes={drawerClasses}
        variant="temporary"
        open={props.isOpen}
        onClose={onClose}
      >
        {content}
      </MuiDrawer>
    )
  )
  return <div className={classes.children}>{drawer}{props.children}</div>
}
Drawer.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node,
}