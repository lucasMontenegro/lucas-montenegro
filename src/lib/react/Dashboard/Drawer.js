import { makeStyles } from "@material-ui/core/styles"
import useTranslation from "lib/react/useTranslation"
import useMediaQuery from "@material-ui/core/useMediaQuery"
import React, { Fragment } from "react"
import AppBar from "@material-ui/core/AppBar"
import Toolbar from "@material-ui/core/Toolbar"
import CloseButton from "./CloseButton"
import AccessButton from "lib/react/AccessButton"
import Settings from "lib/react/Settings"
import SettingsButton from "./SettingsButton"
import Divider from "@material-ui/core/Divider"
import Nav from "./Nav"
import MuiDrawer from "@material-ui/core/Drawer"
const desktopBreakpoint = theme => theme.breakpoints.up(`md`)
const useStyles = makeStyles(theme => ({
  appBar: {
    color: theme.palette.text.primary,
    backgroundColor: theme.palette.type === `dark` ? `#1d1d1dc4` : `#ffffffc4`,
    width: theme.spacing(32),
    left: 0,
    right: `auto`,
    paddingRight: `0px !important`,
  },
  toolbar: {
    justifyContent: `space-between`,
    "& > button.icon-button": {
      minWidth: theme.spacing(6),
    },
  },
  paper: {
    width: theme.spacing(32) + 1,
    paddingTop: theme.spacing(6),
    paddingLeft: 0,
    paddingRight: 0,
    [theme.breakpoints.up(`sm`)]: {
      paddingTop: theme.spacing(8),
    },
  },
  children: {
    [desktopBreakpoint(theme)]: {
      marginLeft: theme.spacing(32) + 1,
    },
  },
}), { name: `lib-react-dashboard` })
export default function Drawer (props) {
  const { onClose } = props
  const classes = useStyles()
  const t = useTranslation()
  const isDesktop = useMediaQuery(desktopBreakpoint)
  const content = (
    <Fragment>
      <AppBar elevation={0} className={classes.appBar}>
        <Toolbar className={classes.toolbar}>
          <CloseButton isDesktop={isDesktop} t={t} onClick={onClose} />
          <AccessButton closeDashboard={onClose} />
          <Settings>{open => <SettingsButton t={t} onClick={open} />}</Settings>
        </Toolbar>
      </AppBar>
      <Divider />
      <Nav t={t} onClick={onClose} />
    </Fragment>
  )
  const drawerClasses = { paper: classes.paper }
  const drawer = (
    isDesktop ? <MuiDrawer classes={drawerClasses} variant="permanent">{content}</MuiDrawer> :
    (
      <MuiDrawer
        id="lib-react-dashboard"
        ModalProps={{ keepMounted: true }}
        classes={drawerClasses}
        variant="temporary"
        open={props.isOpen}
        onClose={onClose}
      >
        {content}
      </MuiDrawer>
    )
  )
  return <div className={classes.children}>{drawer}{props.children}</div>
}