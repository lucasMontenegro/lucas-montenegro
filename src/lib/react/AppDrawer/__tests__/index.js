import React, { createContext, useState } from "react"
import Drawer from "../Drawer"
import renderer from "react-test-renderer"
jest.mock(`react`, () => {
  const React = jest.requireActual("react")
  return {
    ...React,
    __esModule: true,
    default: React,
    createContext: jest.fn(),
    useContext: x => x,
    useState: jest.fn(),
    useCallback: x => x,
  }
})
const AppDrawerProvider = jest.fn(props => (
  <div {...props} className="AppDrawerContext.Provider" />
))
createContext.mockReturnValueOnce({ Provider: AppDrawerProvider })
jest.mock(`../Drawer`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: jest.fn(props => <div {...props} className="Drawer" />) }
})
describe(`../index.js`, () => {
  let AppDrawerContext, useAppDrawerOpener, AppDrawer
  beforeAll(() => {
    ({ AppDrawerContext, useAppDrawerOpener, default: AppDrawer } = require("../index.js"))
  })
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([`react`, `prop-types`])).toMatchSnapshot()
  })
  describe(`AppDrawerContext`, () => {
    it(`should have a default value that runs`, () => {
      expect(createContext.mock.calls).toHaveLength(1)
      expect(createContext.mock.calls[0]).toHaveLength(1)
      const openAppDrawer = createContext.mock.calls[0][0]
      expect(openAppDrawer).toBeInstanceOf(Function)
      openAppDrawer()
    })
  })
  describe(`useAppDrawerOpener`, () => {
    it(`should run`, () => {
      useAppDrawerOpener()
    })
  })
  describe.each([[true], [false]])(`<AppDrawer /> (is open %j)`, isOpen => {
    const setIsOpen = jest.fn()
    let html
    beforeAll(() => {
      useState.mockReturnValueOnce([isOpen, setIsOpen])
      html = renderer.create(<AppDrawer windowTitle="Window Title">Client</AppDrawer>)
    })
    it(`should render`, () => {
      expect(html.toJSON()).toMatchSnapshot()
    })
    describe(`"onClose" drawer event handler`, () => {
      it(`should close the drawer`, () => {
        expect(Drawer.mock.calls).toHaveLength(1)
        const fn = Drawer.mock.calls[0][0].onClose
        expect(fn).toBeInstanceOf(Function)
        fn()
        Drawer.mockClear()
        expect(setIsOpen.mock.calls).toEqual([[false]])
        setIsOpen.mockClear()
      })
    })
    describe(`context value`, () => {
      it(`should open the drawer`, () => {
        expect(AppDrawerProvider.mock.calls).toHaveLength(1)
        const fn = AppDrawerProvider.mock.calls[0][0].value
        expect(fn).toBeInstanceOf(Function)
        fn()
        AppDrawerProvider.mockClear()
        expect(setIsOpen.mock.calls).toEqual([[true]])
        setIsOpen.mockClear()
      })
    })
  })
})