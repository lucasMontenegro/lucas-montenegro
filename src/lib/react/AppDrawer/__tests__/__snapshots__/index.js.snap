// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../index.js <AppDrawer /> (is open false) should render 1`] = `
<div
  className="Drawer"
  isOpen={false}
  onClose={[Function]}
  windowTitle="Window Title"
>
  <div
    className="AppDrawerContext.Provider"
    value={[Function]}
  >
    Client
  </div>
</div>
`;

exports[`../index.js <AppDrawer /> (is open true) should render 1`] = `
<div
  className="Drawer"
  isOpen={true}
  onClose={[Function]}
  windowTitle="Window Title"
>
  <div
    className="AppDrawerContext.Provider"
    value={[Function]}
  >
    Client
  </div>
</div>
`;

exports[`../index.js should use the right versions of its dependencies 1`] = `
Object {
  "prop-types": "^15.7.2",
  "react": "^16.10.1",
}
`;
